test/controller/beast.test.ts

```diff
@@ -0,0 +1,113 @@
+import { Request, Response } from 'express';
+import BeastController from '../../src/controller/beast';
+import BeastModel from '../../src/models/beast';
+import { BadRequestError, UnknownError } from '../../src/models/errors';
+
+jest.mock('../../src/models/beast');
+
+const mockResponse = () => {
+    const res = {} as Response;
+    res.status = jest.fn().mockReturnValue(res);
+    res.json = jest.fn().mockReturnValue(res);
+    return res;
+};
+
+const mockRequest = (body: Record<string, any> = {}, params: Record<string, any> = {}) => ({
+    body,
+    params,
+}) as Request;
+
+// Mock the BeastModel methods
+jest.mock('../../src/models/beast', () => ({
+    create: jest.fn(),
+    getBeastById: jest.fn(),
+    updateBeast: jest.fn(),
+}));
+
+describe('BeastController', () => {
+    describe('create', () => {
+        it('successfully creates a beast', async () => {
+            const req = mockRequest({ gamerTag: 'TestTag', email: 'test@example.com' });
+            const res = mockResponse();
+
+            (BeastModel.create as jest.Mock).mockResolvedValue({ gamerTag: 'TestTag', email: 'test@example.com' });
+
+            await BeastController.create(req, res);
+
+            expect(res.status).toHaveBeenCalledWith(201);
+            expect(res.json).toHaveBeenCalledWith(expect.objectContaining({ message: 'Successfully created' }));
+        });
+
+        it('handles CustomError when creation fails', async () => {
+            const req = mockRequest({ gamerTag: 'TestTag', email: 'test@example.com' });
+            const res = mockResponse();
+
+            (BeastModel.create as jest.Mock).mockRejectedValue(new BadRequestError('Creation failed'));
```

Comment:
Why is this failing? Also do you think it'd be beneficial to have more specific error messages instead of 'Creation failed'? 

It might be useful to be more descriptive in the 'it' statements - why should creation fail?

Reply:
I think there are more potential edge cases we should test for (like creating a user with a username that already exists, etc)